pymeasure/instruments/keithley/keithleyDMM6500.py

```diff
@@ -0,0 +1,1432 @@
+        if "{function}" in command:
+            command = command.replace("{function}", ScannerCard2000Channel.MODES[self.mode])
+        super().write(command)
+
+
+class KeithleyDMM6500(Instrument):
+    """Represents the Keithely DMM6500 6½-Digit Multimeter and provides a
+    high-level interface for interacting with the instrument.
+    This class only uses "SCPI" command set (see also :attr:`command_set`) to
+    communicate with the instrument.
+
+    .. code-block:: python
+
+        # Access via LAN
+        ip_address = "xxx.xxx.xxx.xxx"
+        dmm = KeithleyDMM6500(f"TCPIP::{ip_address}::inst0::INSTR")
+
+
+    User can also use PyVISA to get DMM6500's USB port name and pass it to :class:`KeithleyDMM6500`
+
+    .. code-block:: python
+
+        import pyvisa
+        rm = pyvisa.ResourceManager()
+        resources = rm.list_resources()
+
+        # assume there is only one USB instruments
+        # Ex. ('USB0::1510::25856::01234567::0::INSTR')
+
+        dmm = KeithleyDMM6500( resources[0] )
+
+        # Measure voltage
+        dmm.measure_voltage()
+        print(dmm.voltage)
+
+        # Measure AC voltage
+        dmm.measure_voltage(ac=True)
+        print(dmm.voltage)
+    """
+
+    MODES = {
+        "voltage": "VOLT:DC",
+        "voltage ac": "VOLT:AC",
+        "current": "CURR:DC",
+        "current ac": "CURR:AC",
+        "resistance": "RES",
+        "resistance 4W": "FRES",
+        "diode": "DIOD",
+        "capacitance": "CAP",
+        "temperature": "TEMP",
+        "continuity": "CONT",
+        "period": "PER:VOLT",
+        "frequency": "FREQ:VOLT",
+        "voltage ratio": "VOLT:DC:RAT",
+    }
+
+    MODES_HAVE_AUTORANGE = (
+        "current",
+        "current ac",
+        "voltage",
+        "voltage ac",
+        "resistance",
+        "resistance 4W",
+        "capacitance",
+        "voltage ratio",
+    )
+
+    BOOL_MAPPINGS = {True: 1, False: 0}
+
+    channels = Instrument.MultiChannelCreator(ScannerCard2000Channel, list(range(1, 11)))
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(adapter, "Keithley DMM6500 6½-Digit Multimeter", **kwargs)
+        self.adapter.connection.read_termination = "\n"
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """With context manager, this will fully close the connection."""
+        self.adapter.close()
+
+    def close(self):
+        """Fully closes the connection"""
+        self.adapter.close()
+
+    ###########
+    # General #
+    ###########
+
+    command_set = Instrument.control(
+        "*LANG?",
+        "*LANG %s",
+        """ A string property that controls the command set, which can take the values:
+        :code:`SCPI`, :code:`TSP`, :code:`SCPI2000`, and :code:`SCPI34401`.
+        The :attr:`KeithleyDMM6500` class was designed to use :code:`SCPI` command set only.
+        """,
+        validator=strict_discrete_set,
+        values=["TSP", "SCPI", "SCPI2000", "SCPI34401"],
+    )
```

Comment:
Is it useful to have a command, which basically breaks the instrument for this driver?
Can you return to SCPI mode by using this command, even though the currenct command set is something else?

Reply:
I added a code to force use "SCPI" command set in the `__init__()` to prevent the current command set is something else. I prefer to keep set command if someone really want to change the command set.

Reply:
I can confirm that this command works even if the current command set is TSP or anything else. So I add this command in the init to make sure the instrument uses the SCPI command set from the beginning.